articles/azure-functions/functions-infrastructure-as-code.md

```diff
@@ -305,78 +350,172 @@ The Premium plan offers the same scaling as the Consumption plan but includes de
 
 ### Create a Premium plan
 
-A Premium plan is a special type of "serverfarm" resource. You can specify it by using either `EP1`, `EP2`, or `EP3` for the `Name` property value in the `sku` [description object](/azure/templates/microsoft.web/2018-02-01/serverfarms#skudescription-object).
+A Premium plan is a special type of "serverfarm" resource. You can specify it by using either `EP1`, `EP2`, or `EP3` for the `Name` property value in the `sku` as following:
+
+# [Windows](#tab/Windows)
 
 ```json
 {
-    "type": "Microsoft.Web/serverfarms",
-    "apiVersion": "2018-02-01",
+  "type": "Microsoft.Web/serverfarms",
+  "apiVersion": "2021-02-01",
+  "name": "[variables('hostingPlanName')]",
+  "location": "[parameters('location')]",
+  "sku": {
+    "tier": "ElasticPremium",
+    "name": "EP1",
+    "family": "EP"
+  },
+  "properties": {
     "name": "[parameters('hostingPlanName')]",
-    "location": "[resourceGroup().location]",
-    "properties": {
-        "name": "[parameters('hostingPlanName')]",
-        "workerSize": "[parameters('workerSize')]",
-        "workerSizeId": "[parameters('workerSizeId')]",
-        "numberOfWorkers": "[parameters('numberOfWorkers')]",
-        "hostingEnvironment": "[parameters('hostingEnvironment')]",
-        "maximumElasticWorkerCount": "20"
-    },
-    "sku": {
-        "Tier": "ElasticPremium",
-        "Name": "EP1"
-    }
+    "maximumElasticWorkerCount": 20
+  },
+  "kind": "elastic"
+}
+```
+
+# [Linux](#tab/Linux)
+
+To run your app on Linux, you must also set property `"reserved": true` for the serverfarms resource:
+
+```json
+{
+  "type": "Microsoft.Web/serverfarms",
+  "apiVersion": "2021-02-01",
+  "name": "[variables('hostingPlanName')]",
+  "location": "[parameters('location')]",
+  "sku": {
+    "tier": "ElasticPremium",
+    "name": "EP1",
+    "family": "EP"
+  },
+  "properties": {
+    "name": "[parameters('hostingPlanName')]",
+    "maximumElasticWorkerCount": 20,
+    "reserved": true
+  },
+  "kind": "elastic"
 }
 ```
+---
 
 ### Create a function app
 
-A function app on a Premium plan must have the `serverFarmId` property set to the resource ID of the plan created earlier. In addition, a Premium plan requires an extra setting in the site configuration: [`WEBSITE_CONTENTAZUREFILECONNECTIONSTRING`](functions-app-settings.md#website_contentazurefileconnectionstring). This property configures the storage account where the function app code and configuration are stored.
+For function app on a Premium plan, you will need to set the `serverFarmId` property on the app so that it points to the resource ID of the plan. You should ensure that the function app has a `dependsOn` setting for the plan as well.
```

Comment:
```suggestion
For a function app on a Premium plan, you must set the `serverFarmId` property on the app so that it points to the resource ID of the plan. Make sure that the function app has a `dependsOn` setting that also references the plan.
```